From 872163e123b52a384221510670a0f23bc0ee5ac3 Mon Sep 17 00:00:00 2001
Message-Id: <872163e123b52a384221510670a0f23bc0ee5ac3.1415245452.git.chang-joon.lee@intel.com>
In-Reply-To: <76203ad7c635d495ef3ec1e07b47bad9db4745ba.1415245452.git.chang-joon.lee@intel.com>
References: <76203ad7c635d495ef3ec1e07b47bad9db4745ba.1415245452.git.chang-joon.lee@intel.com>
From: Thomas Daniel <thomas.daniel@intel.com>
Date: Wed, 20 Aug 2014 16:24:50 +0100
Subject: [PATCH 21/30] FROM_UPSTREAM [VPG]: drm/i915/bdw: Do not initialize
 PPGTT in the legacy way for execlists

A pending commit removes synchronous mode from switch_mm.  This breaks
execlists because switch_mm will always try to write to the legacy ring
buffer.

Return immediately from i915_ppgtt_init_gw in execlists mode.
No longer check for execlists mode in gen8_ppgtt_enable() because this
will no longer be called in execlists mode.

Signed-off-by: Thomas Daniel <thomas.daniel@intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
(cherry picked from commit 671b50134ccd75a5dd1584e306a9316587371af3)

Change-Id: I433ed9fa695533b1c374d7ac90c53f4d9e138fde
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: 671b50134ccd75a5dd1584e306a9316587371af3
Upstream-as-of-tag: drm-intel-next-2014-09-05
For: GMINL-2586
For: VIZ-2020
Signed-off-by: Dave Gordon <david.s.gordon@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_gtt.c |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_gtt.c b/drivers/gpu/drm/i915/i915_gem_gtt.c
index 6c5ca41..a313696 100644
--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -873,12 +873,6 @@ static void gen8_ppgtt_enable(struct drm_device *dev)
 	struct intel_engine_cs *ring;
 	int j;
 
-	/* In the case of execlists, PPGTT is enabled by the context descriptor
-	 * and the PDPs are contained within the context itself.  We don't
-	 * need to do anything here. */
-	if (i915.enable_execlists)
-		return 0;
-
 	for_each_ring(ring, dev_priv, j) {
 		I915_WRITE(RING_MODE_GEN7(ring),
 			   _MASKED_BIT_ENABLE(GFX_PPGTT_ENABLE));
@@ -1223,6 +1217,12 @@ int i915_ppgtt_init_hw(struct drm_device *dev)
 	struct i915_hw_ppgtt *ppgtt = dev_priv->mm.aliasing_ppgtt;
 	int i, ret = 0;
 
+	/* In the case of execlists, PPGTT is enabled by the context descriptor
+	 * and the PDPs are contained within the context itself.  We don't
+	 * need to do anything here. */
+	if (i915.enable_execlists)
+		return 0;
+
 	if (!USES_PPGTT(dev))
 		return 0;
 
-- 
1.7.9.5

